hw/dv/tools/dvsim/sim.mk

```
@@ -88,38 +83,46 @@ ifneq (${sw_images},)
-				echo "Building \"$${bazel_label}\" on network connected machine."; \
-				bazel_cmd="./bazelisk.sh"; \
-			else \
-				echo "Building \"$${bazel_label}\" on air-gapped machine."; \
-				bazel_opts+=" --distdir=$${BAZEL_DISTDIR} --repository_cache=$${BAZEL_CACHE}"; \
-				bazel_cmd="bazel"; \
-			fi; \
-			echo "Building with command: $${bazel_cmd} build $${bazel_opts} $${bazel_label}"; \
-			$${bazel_cmd} build $${bazel_opts} $${bazel_label}; \
-			for dep in $$($${bazel_cmd} cquery "labels(data, $${bazel_label})" \
+			target_labels+=$${bazel_target}+" "; \
+		fi; \
+	done; \
+	if [[ ! -z $${target_labels} ]]; then \
+		echo "Building Targets: \"$${target_labels}\"."; \
+		bazel_opts="${sw_build_opts} --define DISABLE_VERILATOR_BUILD=true"; \
+		if [[ -z $${BAZEL_PYTHON_WHEELS_REPO} ]]; then \
+			echo "on network connected machine."; \
+			bazel_cmd="./bazelisk.sh"; \
+		else \
+			echo "on an air-gapped machine."; \
+			bazel_opts+=" --distdir=$${BAZEL_DISTDIR} --repository_cache=$${BAZEL_CACHE}"; \
+			bazel_cmd="bazel"; \
+		fi; \
+		# After the images are built, we use `bazel cquery ...` to locate the built
+		# software artifacts so they can be copied to the test bench run directory.
+		# We only copy device SW images, and do not copy host-side artifacts (like
+		# opentitantool) that are also dependencies of the Bazel test target that
+		# encode the software image targets.
+		echo "Building with command: $${bazel_cmd} build $${bazel_opts} $${target_labels}"; \
+		$${bazel_cmd} build $${bazel_opts} $${target_lables}; \
+		for dep in $$($${bazel_cmd} cquery "labels(data, $${target_labels})" \
+			--ui_event_filters=-info \
+			--noshow_progress \
+			--output=starlark); do \
+			if [[ $$dep != //hw* ]] && [[ $$dep != //util* ]] && [[ $$dep != //sw/host* ]]; then \
+				for artifact in $$($${bazel_cmd} cquery $${dep} \
 				--ui_event_filters=-info \
 				--noshow_progress \
-				--output=starlark); do \
-				if [[ $$dep != //hw* ]] && [[ $$dep != //util* ]] && [[ $$dep != //sw/host* ]]; then \
-					for artifact in $$($${bazel_cmd} cquery $${dep} \
-						--ui_event_filters=-info \
-						--noshow_progress \
-						--output=starlark \
-						--starlark:expr="\"\\n\".join([f.path for f in target.files.to_list()])"); do \
-						cp -f $${artifact} $${run_dir}/$$(basename $${artifact}); \
-						if [[ $$artifact == *.scr.vmem ]]; then \
-							cp -f "$$(echo $${artifact} | cut -d. -f 1).elf" \
-								$${run_dir}/$$(basename "$${artifact%.*.scr.vmem}.elf"); \
-						fi; \
-					done; \
-				fi; \
-			done; \
-		fi; \
-	done;
+				--output=starlark \
+				--starlark:expr="\"\\n\".join([f.path for f in target.files.to_list()])"); do \
+					cp -f $${artifact} $${run_dir}/$$(basename $${artifact}); \
+					if [[ $$artifact == *.scr.vmem ]]; then \
+						cp -f "$$(echo $${artifact} | cut -d. -f 1).elf" \
+						$${run_dir}/$$(basename "$${artifact%.*.scr.vmem}.elf"); \
+					fi; \
+				done; \
+			fi; \
+		done; \
+	fi; \
```

no `\` on the last line here like before:
```suggestion
	fi;
```